ReactSkia/sdk/CurlNetworking.cpp

```diff
@@ -319,14 +334,25 @@ bool CurlNetworking::sendRequest(shared_ptr<CurlRequest> curlRequest, folly::dyn
       RNS_LOG_ERROR ("Not supported method\n" << curlRequest->method) ;
       goto safe_return;
   }
-  curl_multi_add_handle(curlMultihandle_, curl);
-  sem_post(&networkRequestSem_);
+  {
+    std::scoped_lock lock(curlInstanceMutex_);
+    curl_multi_add_handle(curlMultihandle_, curl);
+  }
+  sem_getvalue(&networkRequestSem_, &semCount);
+  if(!semCount){ //We don't have to send signal if count is already 1. Using this as binary sem
+    sem_post(&networkRequestSem_);
+  }
+
   status = true;
   safe_return :
   return status;
 } 
+
 bool CurlNetworking::abortRequest(shared_ptr<CurlRequest> curlRequest) {
   if(curlRequest->handle) {
+    std::scoped_lock lock(curlInstanceMutex_);
```

Comment:
abortRequest when called from mutation processing, will be blocked until curl_multi_perform returns. Hope its is verified that main thread will not be blocked for long time